sambacc/commands/skips.py

```diff
@@ -0,0 +1,188 @@
+#
+
+from typing import Optional
+import argparse
+import os
+
+from sambacc.typelets import Self
+
+from .cli import Context
+
+
+class SkipIf:
+    """Base class for objects used to check if a particular sambacc command
+    should be skipped.
+    Skips are useful when different commands are chained together
+    unconditionally in a configuration file (like k8s init containers) but
+    certain commmands should not be run.
+    """
+
+    NAME: str = ""
+
+    def test(self, ctx: Context) -> Optional[str]:
+        """Return a string explaining the reason for the skip or None
+        indicating no skip is desired.
+        """
+        raise NotImplementedError()  # pragma: nocover
+
+    @classmethod
+    def parse(cls, value: str) -> Self:
+        """Parse a string into a skip class arguments."""
+        raise NotImplementedError()  # pragma: nocover
+
+
+class SkipFile(SkipIf):
+    """Skip execution if a file is, or is not, present. For example,
+    "file:/foo/bar" will skip if the file /foo/bar exists. To skip if a file
+    does not exist, use "file:!/foo/bar" - prefix the file name with an
+    exclaimation point.
+    """
+
+    NAME: str = "file"
+    inverted: bool = False
+    path: str = ""
+
+    @classmethod
+    def parse(cls, value: str) -> Self:
+        obj = cls()
+        if not value:
+            raise ValueError("missing path")
+        if value[0] == "!":
+            obj.inverted = True
+            value = value[1:]
+        obj.path = value
+        return obj
+
+    def test(self, ctx: Context) -> Optional[str]:
+        exists = os.path.exists(self.path)
+        if self.inverted and not exists:
+            return f"skip-if-file-missing: {self.path} missing"
+        if not self.inverted and exists:
+            return f"skip-if-file-exists: {self.path} exists"
+        return None
+
+
+class SkipEnv(SkipIf):
+    """Skip execution if an environment variable is, or is not, equal to a
+    value. The specification is roughly "env:<ENV_VAR><op><VALUE>" where op may
+    be either `==` or `!=`. For example, "env:FLAVOR==cherry" will skip
+    execution if the environment variable "FLAVOR" contains the value "cherry".
+    "env:FLAVOR!=cherry" will skip execution if "FLAVOR" contains any value
+    other than "cherry".
+    """
+
+    NAME: str = "env"
+    _EQ = "=="
+    _NEQ = "!="
+
+    def __init__(self, op: str, var_name: str, value: str) -> None:
+        self.op = op
+        self.var_name = var_name
+        self.target_value = value
+
+    @classmethod
+    def parse(cls, value: str) -> Self:
+        if cls._EQ in value:
+            op = cls._EQ
+        elif cls._NEQ in value:
+            op = cls._NEQ
+        else:
+            raise ValueError("invalid SkipEnv: missing or invalid operation")
+        lhv, rhv = value.split(op, 1)
+        return cls(op, lhv, rhv)
+
+    def test(self, ctx: Context) -> Optional[str]:
+        env_val = os.environ.get(self.var_name)
```

Comment:
I prefer os.environ as it feels more pythonic to have a "special" dict and use typical dict methods rather than relying on specific calls unique to the environment variable handling like C.